cloudsql-postgresql-plugin/src/e2e-test/java/io/cdap/plugin/cloudsqlpostgresql/BQValidation.java

```diff
@@ -0,0 +1,268 @@
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.sql.*;
+import java.time.*;
+import java.util.Date;
+import java.text.ParseException;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+    static List<JsonObject> BigQueryResponse = new ArrayList<>();
+    static List<Object> bigQueryRows = new ArrayList<>();
+
+    /**
+     * Extracts entire data from source and target tables.
+     *
+     * @param sourceTable table at the source side
+     * @param targetTable table at the sink side
+     * @return true if the values in source and target side are equal
+     */
+    public static boolean validateDBToBQRecordValues(String schema, String sourceTable, String targetTable)
+            throws SQLException, ClassNotFoundException, IOException, InterruptedException {
+        getBigQueryTableData(targetTable, bigQueryRows);
+        for (Object rows : bigQueryRows) {
+            JsonObject json = new Gson().fromJson(String.valueOf(rows), JsonObject.class);
+            BigQueryResponse.add(json);
+        }
+        String getSourceQuery = "SELECT * FROM " + schema + "." + sourceTable;
+        try (Connection connect = CloudSqlPostgreSqlClient.getCloudSqlConnection()) {
+            connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+            Statement statement1 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                    ResultSet.HOLD_CURSORS_OVER_COMMIT);
+
+            ResultSet rsSource = statement1.executeQuery(getSourceQuery);
+            return compareResultSetandJsonData(rsSource, BigQueryResponse);
+        }
+    }
+    public static boolean validateBQToDBRecordValues(String schema, String sourceTable, String targetTable)
+            throws SQLException, ClassNotFoundException, IOException, InterruptedException {
+        getBigQueryTableData(sourceTable, bigQueryRows);
+        for (Object rows : bigQueryRows) {
+            JsonObject json = new Gson().fromJson(String.valueOf(rows), JsonObject.class);
+            BigQueryResponse.add(json);
+        }
+        String getTargetQuery = "SELECT * FROM " + schema + "." + targetTable;
+        try (Connection connect = CloudSqlPostgreSqlClient.getCloudSqlConnection()) {
+            connect.setHoldability(ResultSet.HOLD_CURSORS_OVER_COMMIT);
+            Statement statement1 = connect.createStatement(ResultSet.TYPE_SCROLL_SENSITIVE, ResultSet.CONCUR_UPDATABLE,
+                    ResultSet.HOLD_CURSORS_OVER_COMMIT);
+
+            ResultSet rsTarget = statement1.executeQuery(getTargetQuery);
+            return compareResultSetandJsonData(rsTarget, BigQueryResponse);
+        }
+    }
+    /**
+     * Retrieves the data from a specified BigQuery table and populates it into the provided list of objects.
+     *
+     * @param table        The name of the BigQuery table to fetch data from.
```

Comment:
Remove the extra spacing.

Reply:
Done :+1:

Reply:
I don't see any commit. push the commit and add the link here

Reply:
https://github.com/cloudsufi/database-plugins/pull/12/commits/351fee60e134bec0d4e1019acad4c9bb3259952b